### DISCUSSION TITLE
TUI in Python

### DISCUSSION THREAD

**Discussion by pawamoy** [git-bug/git-bug#1107] [Q&A] [answered]
Hello!

How would I go about writing a TUI for git-bug in Python :slightly_smiling_face:?
Is there a way to query data from git-bug in a language agnostic way?
Should I just go read the files it writes in the repo, following the [docs on data models](https://github.com/MichaelMure/git-bug/blob/master/doc/model.md)?

I'd use [Textual](https://github.com/Textualize/textual) as TUI framework, so all I'm asking here is how I can *read data from git-bug* :slightly_smiling_face: Any hint is appreciated!

**Comment by MichaelMure** [comment, marked as answer]
The best way would be the graphql API. That way you would not need to reimplement git-bug from zero (which is far from trivial), and you would not need to follow the future changes.

Not to discourge you from that adventure, but why not contribute to git-bug directly?

**Reply by pawamoy** [reply]
I have no experience with Go, and am super comfortable in Python, and Textual is amazing :D 
I suppose it is an indirect contribution if more people get to know/like git-bug thanks to a third-party TUI (getting ahead of myself) :)

---

OK thanks, GraphQL seems the way to go!
